### PR TITLE
Wrap an arrow function in parentheses if it the test of a conditional expression

### LINKED ISSUE #5819: [bug] Arrow function in ternary test should be wrapped in parentheses
### Input Code
```js
(() => {}) ? 0 : 1
```

https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=&targets=&browsers=&builtIns=false&debug=false&code=(()%20%3D%3E%20%7B%7D)%20%3F%200%20%3A%201

### Babel Configuration (.babelrc, package.json, cli command)

```js
{}
```

### Expected Behavior
The code should not be modified

### Current Behavior
```js
() => {} ? 0 : 1
```

### Possible Solution
I will open a pr soon

### Context
I found this while working on https://github.com/babel/babylon/pull/559

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | yes
| Major: Breaking Change?  | no
| Minor: New Feature?      | no
| Deprecations?            | no
| Spec Compliancy?         | yes
| Tests Added/Pass?        | yes
| Fixed Tickets            | Fixes #5819
| License                  | MIT
| Doc PR                   | no
| Dependency Changes       | no

---

```js
() => {} ? () => {} : () => {}
```
The first function must be wrapped in parentheses. This pr adds a check to `babel-generator/.../parentheses.js@ArrowFunction` so that it returns `true` in that case.

### CODE REVIEW

File: packages/babel-generator/src/node/parentheses.js [hunk 1]
```diff
@@ -138,6 +138,7 @@ export function ArrowFunctionExpression(node: Object, parent: Object): boolean {
     t.isBinaryExpression(parent) ||
     t.isLogicalExpression(parent) ||
     t.isUnaryExpression(parent) ||
+    (t.isConditionalExpression(parent) && parent.test === node) ||
```
Comment (jridgewell) [inline packages/babel-generator/src/node/parentheses.js]: Nit: `t.isConditionalExpression(parent, { test: node })`.

**Comment by nicolo-ribaudo** [comment]
I didn't use `ConditionalExpresion` because I thought that, like `yield`, `await () => {}` didn't need parentheses.

Using `ConditionalExpression` I noticed that that function didn't check if `t.isTaggedTemplateExpression(parent)`

**Comment by nicolo-ribaudo** [comment]
I was wonderin

**Comment by nicolo-ribaudo** [comment]
I was wondering if `t.isExportStatement(parent)` can be removed: it should only be needed to disambiguate between function expressions and function statements (thus the original `export default (function () {})` comment), but arrow functions can't be misinterpreted as statements.

**Comment by hzoo** [comment]
Nice work @nicolo-ribaudo!!

**Pull request merged**